.github/workflows/build-push.yml

```diff
@@ -34,44 +36,22 @@ jobs:
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Extract branch name
-        id: extract_branch
-        shell: bash
-        run: |
-          # Extract branch name
-          BRANCH_NAME=${GITHUB_REF#refs/heads/}
-
-          # set output variable accessible in the action
-          echo ::set-output name=branch::${BRANCH_NAME}
-
-      # For debugging capture the selected branch
-      - name: Extracted branch
-        run: echo "Extracted branch ${{ steps.extract_branch.outputs.branch }}"
+      - id: extract_branch
+        uses: swapActions/get-branch-name@v1
 
       - name: Build and push
         id: docker_build
-        uses: docker/build-push-action@v2
+        uses: docker/build-push-action@v3
         with:
           push: true
           tags: |
             dmsc/duo-gateway:${{ steps.extract_branch.outputs.branch }}
             ghcr.io/userofficeproject/user-office-gateway:${{ steps.extract_branch.outputs.branch }}
 
       - name: Trigger pipeline
-        run: |
-          REPOSITORY=${{ github.repository }}
-          RESULT=$(curl -X POST \
-            -F token="${{ secrets.GITLAB_TRIGGER_TOKEN }}" \
-            -F ref=master \
-            -F "variables[RUN_DEPLOY]=1" \
-            -F "variables[BRANCH]=${{ steps.extract_branch.outputs.branch }}" \
-            -F "variables[REPOSITORY]=${REPOSITORY#UserOfficeProject/}" \
-            -o /dev/null \
-            --silent \
-            --write-out '%{http_code}' \
-            ${{ secrets.GITLAB_TRIGGER_URL }})
-
-          if [[ $RESULT != "201" ]]; then
-            echo "Failed to trigger Gitlab pipeline: $RESULT"
-            exit 1
-          fi
+        uses: swapActions/trigger-useroffice-deployment@v1
```

Comment:
I like this reusable action to be able to trigger the deployment. Previously it was little bit messy and not so readable :) Very nice move.

Reply:
👍🏻